### PR TITLE
Adds improved interface for `useSignTransaction`

### PR DESCRIPTION
**Pull request by obber**

### CODE REVIEW

File: src/hooks/public/use-sign-transaction.tsx [hunk 1]
```diff
@@ -1,135 +1,164 @@
...
+  /** Whether to send the transaction signing request or not. Defaults to `true`. */
+  enabled?: boolean;
+  /** The unsigned transaction as a base-58 string. */
+  unsignedTransactionB58: string | undefined;
+}
+
+interface UseSignTransactionHookReturn {
+  data: string | undefined;
+  error: SignTransactionErrors | undefined;
+  signed: boolean;
```
Comment (ricebin) [inline src/hooks/public/use-sign-transaction.tsx:36]: shouldnt the presence of data implies signed?

File: src/hooks/public/use-sign-transaction.tsx [hunk 2]
```diff
@@ -1,135 +1,164 @@
...
-  const { close, connection, open } = usePopupConnection();
+interface UseSignTransactionParameters {
+  /** Whether to send the transaction signing request or not. Defaults to `true`. */
+  enabled?: boolean;
+  /** The unsigned transaction as a base-58 string. */
+  unsignedTransactionB58: string | undefined;
+}
+
+interface UseSignTransactionHookReturn {
+  data: string | undefined;
```
Comment (ricebin) [inline src/hooks/public/use-sign-transaction.tsx:34]: // then comment this is a transaction signature

File: src/hooks/public/use-sign-transaction.tsx [hunk 3]
```diff
@@ -1,135 +1,164 @@
...
+  error: SignTransactionErrors | undefined;
+  signed: boolean;
+}
+
+export const useSignTransaction = ({
+  enabled = true,
```
Comment (ricebin) [inline src/hooks/public/use-sign-transaction.tsx:40]: per conversation, we dont need this

**Review by ricebin — APPROVED** [review]

File: src/hooks/public/use-sign-transaction.tsx (hunk 2 above)
Comment (obber) [reply src/hooks/public/use-sign-transaction.tsx:33]: Done.

**Pull request merged**